### PR TITLE
Update in GATK4 variantfiltration: Added vcf_index to vcf tuple; output to vcf.gz format.

### PR DESCRIPTION
**Pull request by praveenraj2018**
Added vcf_index to vcf tuple; output to vcf.gz format.

## PR checklist
- [x ] This comment contains a description of changes (with reason).
- [] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [x ] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: modules/gatk4/variantfiltration/main.nf [hunk 1]
```diff
@@ -19,14 +19,14 @@ process GATK4_VARIANTFILTRATION {
     }
 
     input:
-    tuple val(meta), path(vcf)
+    tuple val(meta), path(vcf), path(vcf_index)
```
Comment (FriederikeHanssen) [inline modules/gatk4/variantfiltration/main.nf:22]:
```suggestion
    tuple val(meta), path(vcf), path(vcf_tbi)
```

We discussed trying to standaradise naming on monday and settled on tbi https://github.com/nf-core/modules/pull/1074/files#diff-c0d34d7885b819641586baffb1a7a05dc3832a955f461d3937a2b984522d2c5e

File: modules/gatk4/variantfiltration/main.nf [hunk 2]
```diff
@@ -19,14 +19,14 @@ process GATK4_VARIANTFILTRATION {
...
     path  fasta
     path  fai
     path  dict
 
     output:
-    tuple val(meta), path("*.vcf"), emit: vcf
-    path "versions.yml"           , emit: versions
+    tuple val(meta), path("*.vcf.gz")	, emit: vcf
+    path "versions.yml"           	, emit: versions
```
Comment (FriederikeHanssen) [inline modules/gatk4/variantfiltration/main.nf:29]: Please align here by `,`

File: modules/gatk4/variantfiltration/meta.yml [hunk 3]
```diff
@@ -20,9 +20,10 @@ input:
         Groovy Map containing sample information
         e.g. [ id:'test']
   - vcf:
-      type: list
-      description: Input VCF file
-      pattern: "*.{vcf}"
+      type: tuple of files
+      description: Tuple of VCF(.gz) file (first) and its index (second)
+      pattern: ["*.{vcf,vcf.gz}", "*.{idx,tbi}"]
+
```
Comment (FriederikeHanssen) [inline modules/gatk4/variantfiltration/meta.yml:26]: Entry for `vcf.tbi` missing

File: modules/gatk4/variantfiltration/main.nf (hunk 2 above)
Comment (FriederikeHanssen) [reply modules/gatk4/variantfiltration/main.nf:29]: Just to clariify, no output tbi produced?

File: modules/gatk4/variantfiltration/main.nf [hunk 4]
```diff
@@ -35,7 +35,7 @@ process GATK4_VARIANTFILTRATION {
     gatk VariantFiltration \\
         -R $fasta \\
         -V $vcf \\
-        -O ${prefix}.vcf \\
+        -O ${prefix}.vcf.gz \\
         $options.args
```
Comment (FriederikeHanssen) [inline modules/gatk4/variantfiltration/main.nf:39]: is there any way to specify number of cpus etc.?

**Review by FriederikeHanssen — CHANGES_REQUESTED** [review]
Looks good, just some minor points on naming :)

File: modules/gatk4/variantfiltration/main.nf (hunk 1 above)
Comment (praveenraj2018) [reply modules/gatk4/variantfiltration/main.nf:22]: Sounds good. I can make changes and submit again.

Comment (praveenraj2018) [reply modules/gatk4/variantfiltration/main.nf:22]: Done!

File: modules/gatk4/variantfiltration/main.nf (hunk 2 above)
Comment (praveenraj2018) [reply modules/gatk4/variantfiltration/main.nf:29]: Yes it generates tbi. I added that now

File: modules/gatk4/variantfiltration/meta.yml (hunk 3 above)
Comment (praveenraj2018) [reply modules/gatk4/variantfiltration/meta.yml]: Added

File: modules/gatk4/variantfiltration/main.nf (hunk 4 above)
Comment (praveenraj2018) [reply modules/gatk4/variantfiltration/main.nf:40]: There is a way for the spark version of GATK. May be we can see this later?

**Comment by praveenraj2018** [comment]
@FriederikeHanssen can you see why the lint is failing?

**Comment by FriederikeHanssen** [comment]
Hey @praveenraj2018 ! Linting is failing on all modules right now 😞 see discussion here: https://nfcore.slack.com/archives/CJRH30T6V/p1637167057227600

File: modules/gatk4/variantfiltration/main.nf (hunk 4 above)
Comment (FriederikeHanssen) [reply modules/gatk4/variantfiltration/main.nf:40]: ok that then requires  a different module I would say. One other thing, for the GATK we have these lines generally:

```
def avail_mem       = 3
    if (!task.memory) {
        log.info '[GATK HaplotypeCaller] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
    } else {
        avail_mem = task.memory.giga
    }
```

Some gatk thing I believe to make sure enough memory is available.

Comment (praveenraj2018) [reply modules/gatk4/variantfiltration/main.nf:40]: Ok yes, I can add that as well.

Comment (praveenraj2018) [reply modules/gatk4/variantfiltration/main.nf:40]: @FriederikeHanssen if we use `avail_mem` inside Xmx as below, the test fails. I think it breaks the quotes
`gatk --java-options "-Xmx${avail_mem}g" \\`

I have seen some modules that define the variable, but NOT used anywhere. why is it so? One example [here](https://github.com/nf-core/modules/blob/master/modules/gatk4/markduplicates/main.nf)

Comment (FriederikeHanssen) [reply modules/gatk4/variantfiltration/main.nf:40]: You are right,  it needs to be specified somehow. I will add it to n existing issue on gatk memory. 
Can you try it without quotes like [here](https://github.com/nf-core/atacseq/blob/1b3a832db5a53c92c2c89a4d6d79455e860461ad/main.nf#L739)

Comment (praveenraj2018) [reply modules/gatk4/variantfiltration/main.nf:45]: Just removing quotes did not work. But it is solved with the followed way:

`gatk --java-options "-Xmx${avail_mem}G"`

Recommendation here: https://gatk.broadinstitute.org/hc/en-us/articles/360035531892-GATK4-command-line-syntax
Resubmitted PR

Comment (FriederikeHanssen) [reply modules/gatk4/variantfiltration/main.nf:45]: great 👍

File: modules/gatk4/variantfiltration/main.nf [hunk 5]
```diff
@@ -19,23 +19,29 @@ process GATK4_VARIANTFILTRATION {
...
     path  fai
     path  dict
 
     output:
-    tuple val(meta), path("*.vcf"), emit: vcf
-    path "versions.yml"           , emit: versions
-
+    tuple val(meta), path("*.vcf.gz"), emit: vcf
+    tuple val(meta), path("*.tbi")   , emit: tbi
+    path "versions.yml"		     , emit: versions
```
Comment (FriederikeHanssen) [inline modules/gatk4/variantfiltration/main.nf:30]: Align with line above by `,`

**Review by FriederikeHanssen — CHANGES_REQUESTED** [review]
Loos almost good. just the indentation and I can approve. We just need to wait for the linting to be fixed

File: modules/gatk4/variantfiltration/main.nf (hunk 5 above)
Comment (praveenraj2018) [reply modules/gatk4/variantfiltration/main.nf]: Done!

**Comment by FriederikeHanssen** [comment]
I opened a discussion on slack on how to proceed with the failing lints

**Review by FriederikeHanssen — APPROVED** [review]
linting is fixed now 🥳

**Comment by praveenraj2018** [comment]
Thanks @FriederikeHanssen

**Pull request merged**